stages/org.osbuild.rpm

```diff
@@ -47,16 +45,54 @@ SCHEMA = """
     "description": "Array of RPM content hashes",
     "type": "array",
     "items": {
-      "type": "string",
-      "description": ".rpm file checksum, prefixed with 'md5:', 'sha1:', 'sha256:', 'sha384:', or 'sha512:', indicating the algorithm used."
+      "oneOf": [
+        {
+          "type": "string",
+          "description": ".rpm file checksum, prefixed with 'md5:', 'sha1:', 'sha256:', 'sha384:', or 'sha512:', indicating the algorithm used."
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["checksum"],
+          "properties": {
+            "checksum": {
+              "type": "string",
+              "description": ".rpm file checksum, prefixed with 'md5:', 'sha1:', 'sha256:', 'sha384:', or 'sha512:', indicating the algorithm used."
+            },
+            "check_gpg": {
+              "type": "boolean",
+              "description": "Whether the GPG signatures of the RPM should be verified.",
+              "default": false
+            }
+          }
+        }
+      ]
     }
   }
 }
 """
 
 
+def packages_from_legacy(legacy):
+    packages = []
+    for package in legacy:
+        if isinstance(package, dict):
+            packages.append(package)
+        else:
+            packages.append({"checksum": package, "check_gpg": False})
+    return packages
+
+
+def checksums_from_packages(packages):
+    checksums = []
+    for pkg in packages:
+        checksums.append(pkg["checksum"])
+    return checksums
+
+
 def main(tree, sources, options):
-    packages = options.get("packages", [])
+    packages = packages_from_legacy(options.get("packages", []))
+    checksums = checksums_from_packages(packages)
```

Comment:
This could be expressed in a simpler way than using a helper function:
```python
checksums = [p["checksum"] for p in packages]
```

Reply:
Thanks! Fixed.